pkg/scalers/azure/azure_app_insights.go

```diff
@@ -16,18 +16,27 @@ import (
 )
 
 const (
-	appInsightsResource = "https://api.applicationinsights.io"
+	DefaultAppInsightsResourceURL = "https://api.applicationinsights.io"
 )
 
+var AppInsightsResourceURLInCloud = map[string]string{
+	"AZUREPUBLICCLOUD":       "https://api.applicationinsights.io",
+	"AZUREUSGOVERNMENTCLOUD": "https://api.applicationinsights.us",
+	"AZURECHINACLOUD":        "https://api.applicationinsights.azure.cn",
+}
+
 type AppInsightsInfo struct {
-	ApplicationInsightsID string
-	TenantID              string
-	MetricID              string
-	AggregationTimespan   string
-	AggregationType       string
-	Filter                string
-	ClientID              string
-	ClientPassword        string
+	ApplicationInsightsID   string
+	TenantID                string
+	MetricID                string
+	AggregationTimespan     string
+	AggregationType         string
+	Filter                  string
+	ClientID                string
+	ClientPassword          string
+	Cloud                   string
```

Comment:
I can't see any usage of this field outside the metadata parsing scope, so maybe we could remove this field from the struct and use it locally there.
WDYT?

Reply:
Makes sense, will remove it. Can always add it back in the future if required later.